invenio_records_files/views.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio-Records-Files REST integration."""
+
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint, abort, g
+from invenio_files_rest.views import bucket_view, object_view
+from six import iteritems
+
+from invenio_records_files.models import RecordsBuckets
+
+
+def create_blueprint_from_app(app):
+    """Create Invenio-Records-Files blueprint from a Flask application.
+
+    :params app: A Flask application.
+    :returns: Configured blueprint.
+    """
+    records_files_blueprint = Blueprint(
+        'invenio_records_files',
+        __name__,
+        url_prefix='')
+
+    for config_name, rec_types in \
+            iteritems(app.config['RECORDS_FILES_REST_ENDPOINTS']):
+        files_resource_name = app.config['RECORDS_FILES_ENDPOINT_SUFFIX']
+        for rec_type in rec_types:
+            record_item_path = app.config[config_name][rec_type]['item_route']
+            records_files_blueprint.add_url_rule(
+                '{record_item_path}/{files_resource_name}'.format(**locals()),
+                view_func=bucket_view,
+            )
+
+            records_files_blueprint.add_url_rule(
+                '{record_item_path}/{files_resource_name}/<path:key>'.format(
+                    **locals()),
+                view_func=object_view,
+            )
+
+    @records_files_blueprint.url_value_preprocessor
+    def resolve_pid_to_bucket_id(endpoint, values):
+        """Flask URL preprocessor to resolve pid to Bucket ID.
+
+        In the ``records_files_blueprint`` we are gluing together Records-REST
+        and Files-REST APIs. Records-REST knows about PIDs but Files-REST does
+        not, this function will pre-process the URL so the PID is removed from
+        the URL and resolved to bucket ID which is injected into Files-REST
+        view calls:
+        /api/<record_type>/<pid_value>/files/<key> -> /files/<bucket>/<key>.
+        """
+        uuid = str(values['pid_value'].data[0].object_uuid)
+        # We are removing the value since we need the request values to include
+        # only the ones contained in the view signature.
+        g.pid = values.pop('pid_value')
+        pid, record = g.pid.data
+        if hasattr(record, 'files') and record.files is not None:
+            values['bucket_id'] = str(record.files.bucket.id)
+        else:
+            abort(404, 'The requested record does not have files associated')
```

Comment:
Can I suggest:
`
abort(404, 'The requested record with PID {} does not have files associated'.format(pid))
`
Just thinking that if you get a Sentry error, at least you have a PID to go and check quickly

Reply:
We removed the error message completely since we don't want to expose any information.